runtime/interpreter/encode.go

```diff
@@ -1002,44 +1021,54 @@ func (e *Encoder) encodeCompositeValue(
 	}
 
 	// Encode kind at array index encodedCompositeValueKindFieldKey
-	err = e.enc.EncodeUint(uint(v.Kind))
+	err = e.enc.EncodeUint(uint(v.Kind()))
 	if err != nil {
 		return err
 	}
 
 	// Encode fields (as array) at array index encodedCompositeValueFieldsFieldKey
-	err = e.enc.EncodeArrayHead(uint64(v.Fields.Len() * 2))
-	if err != nil {
-		return err
-	}
 
-	// Pre-allocate and reuse valuePath.
-	valuePath := append(path, "")
-
-	lastValuePathIndex := len(path)
-
-	for pair := v.Fields.Oldest(); pair != nil; pair = pair.Next() {
-		fieldName := pair.Key
-
-		// Encode field name as fields array element
-		err := e.enc.EncodeString(fieldName)
+	// If the fields are not loaded, dump the raw fields content as it is.
+	if v.content != nil {
+		err := e.enc.EncodeRawBytes(v.fieldsContent)
+		if err != nil {
+			return err
+		}
```

Comment:
Should `v.content` be changed to `v.fieldsContent`?

Reply:
Yes, that was a silly mistake. And a test case was missing. Fixed it in c4c2c8b.